### PR TITLE
Remind user to check yaml file before creating a cluster

### LINKED ISSUE #197: No explanation for how to edit yaml files for create cluster
/kind bug
/sig ibmcloud

**What steps did you take and what happened:**
Following doc to create cluster and no details about how to edit cluster.yaml and machines.yaml

**What did you expect to happen:**
Some explanation about values in yaml file.

**Anything else you would like to add:**

**Environment:**

- Cluster-api version: mainline
- Minikube/KIND version:  kind 0.3.0
- Kubernetes version: (use `kubectl version`): 1.14
- OS (e.g. from `/etc/os-release`): Ubuntu

### PR DESCRIPTION
**Pull request by wanghh2000**
**What this PR does / why we need it**:
Correct commands of readme 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #197 

**Specify your PR reviewers**:

/cc @gyliu513 @jichenjc @xunpan

**Special notes for your reviewer**:

/sig ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.
No changes on image

**Release note**:
None

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -179,9 +179,9 @@ Once you have created a cluster, you can interact with the cluster and machine
 resources using kubectl:
 
 ```bash
-kubectl --kubeconfig=kubeconfig get clusters
-kubectl --kubeconfig=kubeconfig get machines
-kubectl --kubeconfig=kubeconfig get machines -o yaml
+kubectl --kubeconfig=$KUBECONFIG get clusters
```
Comment (gyliu513) [inline README.md]: After kind set `KUBECONFIG`, we actually do not need `--kubeconfig=$KUBECONFIG`.

Here the command is a generic command for KIND, MiniKube and existing cluster, can you help think more to update the document work for all of those three cases?

**Comment by xunpan** [comment]
The change seems not correct.
`kubeconfig` is generated by `clusterctl` to access target kubernetes cluster. It is created in the `cwd` of your execution of `clusterctl` when target cluster is created.

The context is
```
- Interacting with your cluster 
Once you have created a cluster, you can interact with the cluster and machine resources using kubectl:
```
It should be operated on target cluster instead of bootstrap cluster. So, I think the change for this should be tell
 - where kubeconfig is? what is it?
 - remove `KUBECONFIG`
 - optionally: add bootstrap cluster access document with `KUBECONFIG`. This is not mandatory for end users to me.

**Comment by jichenjc** [comment]
+1 to indicate where the kubeconfig comes from and where is it ...

**Comment by wanghh2000** [comment]
Yes, agree @xunpan @jichenjc  @gyliu513  comments about the kubeconfig.

Just a little confusion about the step in doc. Try to update doc to make it clear. 
```
export KUBECONFIG="$(kind get kubeconfig-path --name="clusterapi")"
```
Another is that we need some explaination about how to edit cluster.yaml & machines.yaml.

**Comment by gyliu513** [comment]
> Another is that we need some explaination about how to edit cluster.yaml & machines.yaml.

This is exactly we need to update. ;-)

**Comment by gyliu513** [comment]
@clyang82 for comments

File: README.md [hunk 2]
```diff
@@ -131,6 +131,9 @@ kube-system                kubernetes-dashboard-5f7b999d65-ntrxb         1/1
 ```
 
 2. Create a cluster:
+
+You have to [inspect](docs/ibmcloud_help.md#inspect-files-clusteryaml-and-machinesyaml) files `cluster.yaml`, `machines.yaml` before create a cluster on IBM Cloud.
```
Comment (xunpan) [inline README.md]: - `s/create/creating/`
 - `s/have to/should/`
 - `inspect` is a verb, a link should be better for a noun word

File: docs/ibmcloud_help.md [hunk 3]
```diff
@@ -3,12 +3,101 @@
...
   - [How to set correct OS reference code in `machines.yaml`?](#how-to-set-correct-os-reference-code-in-machinesyaml)
-  - [how to set `dataCenter` value in `machines.yaml`?](#how-to-set-datacenter-value-in-machinesyaml)
+  - [How to set `dataCenter` value in `machines.yaml`?](#how-to-set-datacenter-value-in-machinesyaml)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
 # IBM Cloud
+## Inspect files `cluster.yaml` and `machines.yaml`.
+
+cluster.yaml
```
Comment (xunpan) [inline docs/ibmcloud_help.md]: this is `ibmcloud_help.md` which is targeted to IBM Cloud related knowledge.  `cluster.yaml` has no relationship with IBM cloud but for cluster API.

File: docs/ibmcloud_help.md [hunk 4]
```diff
@@ -3,12 +3,101 @@
...
+2. Make sure use the same cluster name in labels.
+3. Use valid SSH Key name and user name.
+```
+apiVersion: "cluster.k8s.io/v1alpha1"
+kind: MachineList
+items:
+- apiVersion: "cluster.k8s.io/v1alpha1"
+  kind: Machine
+  metadata:
+    generateName: ibmcloud-master-                         # Prefix of device name
```
Comment (xunpan) [inline docs/ibmcloud_help.md]: Shall we just add such comments directly in `machines.yaml.template`? It is easy to read by end user. 

And for me, not all comments are is necessary.
E.g. need explaination: `generateName: ibmcloud-master- ` will be VM instance name prefix
E.g. makes less sense: `cluster.k8s.io/cluster-name: "test1"                 # cluser name`, cluster name explaining cluster name makes less sense.

File: docs/ibmcloud_help.md (hunk 3 above)
Comment (wanghh2000) [reply docs/ibmcloud_help.md]: OK, Is the trouble-shooting.md good to tell user about the inconsistent cluster could cause failure?
Seems that new user may hit this failure. No error message about it.

File: docs/ibmcloud_help.md (hunk 4 above)
Comment (wanghh2000) [reply docs/ibmcloud_help.md]: File achines.yaml.template is ok, and remind user to do a inspection for yaml files in ReadMe.md. What do you think?

File: README.md (hunk 2 above)
Comment (wanghh2000) [reply README.md]: Update for doc review.

**Comment by wanghh2000** [comment]
@xunpan @jichenjc @clyang82  Would you please help review again? Thanks for your efforts.

File: README.md [hunk 5]
```diff
@@ -131,6 +131,9 @@ kube-system                kubernetes-dashboard-5f7b999d65-ntrxb         1/1
 ```
 
 2. Create a cluster:
+
+You should review files `cluster.yaml`, `machines.yaml`, and ensure the domain and dataCenter and osReferenceCode are set. If you want to do any customization now is the time to do it.
```
Comment (gyliu513) [inline README.md]: s/domain/`domain`
s/dataCenter/`dataCenter`
s/osReferenceCode/`osReferenceCode`

You should review files `cluster.yaml` and `machines.yaml`, ensure the `domain`, `dataCenter` and `osReferenceCode` are set. You can customize those values based on your requirement.

Comment (wanghh2000) [reply README.md]: Thanks @gyliu513's comment and Addressed.

**Pull request merged**